From 749ad392f18f9040ef54e648e6aff7c044962348 Mon Sep 17 00:00:00 2001
From: Zhan Lou <zhan.lou@cixtech.com>
Date: Tue, 27 Jun 2023 19:51:45 +0800
Subject: [PATCH] DPTSW-2351: v4l2: use v4l2 enum for VP9 profiles

Use v4l2 enum v4l2_mpeg_video_vp9_profile instead of magic numbers for
VP9 profiles.

Signed-off-by: Zhan Lou <zhan.lou@cixtech.com>
Change-Id: Ibf11a98652421666438a6dba98ad525f1317a0a9

diff --git a/sys/v4l2/gstv4l2vp9codec.c b/sys/v4l2/gstv4l2vp9codec.c
index b4f2d5cf04..cb0c65aba0 100644
--- a/sys/v4l2/gstv4l2vp9codec.c
+++ b/sys/v4l2/gstv4l2vp9codec.c
@@ -36,13 +36,13 @@ v4l2_profile_from_string (const gchar * profile)
   gint v4l2_profile = -1;
 
   if (g_str_equal (profile, "0"))
-    v4l2_profile = 0;
+    v4l2_profile = V4L2_MPEG_VIDEO_VP9_PROFILE_0;
   else if (g_str_equal (profile, "1"))
-    v4l2_profile = 1;
+    v4l2_profile = V4L2_MPEG_VIDEO_VP9_PROFILE_1;
   else if (g_str_equal (profile, "2"))
-    v4l2_profile = 2;
+    v4l2_profile = V4L2_MPEG_VIDEO_VP9_PROFILE_2;
   else if (g_str_equal (profile, "3"))
-    v4l2_profile = 3;
+    v4l2_profile = V4L2_MPEG_VIDEO_VP9_PROFILE_3;
   else
     GST_WARNING ("Unsupported profile string '%s'", profile);
 
@@ -53,13 +53,13 @@ static const gchar *
 v4l2_profile_to_string (gint v4l2_profile)
 {
   switch (v4l2_profile) {
-    case 0:
+    case V4L2_MPEG_VIDEO_VP9_PROFILE_0:
       return "0";
-    case 1:
+    case V4L2_MPEG_VIDEO_VP9_PROFILE_1:
       return "1";
-    case 2:
+    case V4L2_MPEG_VIDEO_VP9_PROFILE_2:
       return "2";
-    case 3:
+    case V4L2_MPEG_VIDEO_VP9_PROFILE_3:
       return "3";
     default:
       GST_WARNING ("Unsupported V4L2 profile %i", v4l2_profile);
@@ -75,7 +75,7 @@ gst_v4l2_vp9_get_codec (void)
   static GstV4l2Codec *codec = NULL;
   if (g_once_init_enter (&codec)) {
     static GstV4l2Codec c;
-    c.profile_cid = V4L2_CID_MPEG_VIDEO_VPX_PROFILE;
+    c.profile_cid = V4L2_CID_MPEG_VIDEO_VP9_PROFILE;
     c.profile_to_string = v4l2_profile_to_string;
     c.profile_from_string = v4l2_profile_from_string;
     g_once_init_leave (&codec, &c);
